src/db/collections/ledger_update.rs

```diff
@@ -108,52 +108,30 @@ impl MongoDb {
             )
             .await?;
 
-        collection
-            .create_index(
-                IndexModel::builder()
-                    .keys(doc! { "output_id": 1, "is_spent": 1 })
-                    .options(
-                        IndexOptions::builder()
-                            // An output can be spent and unspent only once.
-                            .unique(true)
-                            .name("id_index".to_string())
-                            .build(),
-                    )
-                    .build(),
-                None,
-            )
-            .await?;
-
         Ok(())
     }
 
     /// Upserts a [`Output`](crate::types::stardust::block::Output) together with its associated
     /// [`OutputMetadata`](crate::types::ledger::OutputMetadata).
     pub async fn insert_ledger_updates(
         &self,
-        outputs_with_metadata: impl IntoIterator<Item = OutputWithMetadata>,
+        deltas: impl IntoIterator<Item = OutputWithMetadata>,
     ) -> Result<(), Error> {
-        // TODO: Use `insert_many` and `update_many` to increase write performance.
-
-        for OutputWithMetadata { output, metadata } in outputs_with_metadata {
-            let at = metadata.spent.map_or(metadata.booked, |s| s.spent);
-            let is_spent = metadata.spent.is_some();
-
+        for delta in deltas {
+            self.insert_output(delta.clone()).await?;
             // Ledger updates
-            for owner in output.owning_addresses() {
-                let ledger_update_document = LedgerUpdateDocument {
+            for owner in delta.output.owning_addresses() {
+                let doc = bson::to_document(&LedgerUpdateDocument {
                     address: owner,
-                    output_id: metadata.output_id,
-                    at,
-                    is_spent,
-                };
-
-                let _ = self
-                    .0
+                    output_id: delta.metadata.output_id,
+                    at: delta.metadata.spent.map(|s| s.spent).unwrap_or(delta.metadata.booked),
+                    is_spent: delta.metadata.spent.is_some(),
+                })?;
+                self.0
                     .collection::<LedgerUpdateDocument>(LedgerUpdateDocument::COLLECTION)
                     .update_one(
-                        doc! { "output_id": metadata.output_id, "is_spent": is_spent },
-                        doc! { "$setOnInsert": bson::to_document(&ledger_update_document)? },
+                        doc.clone(),
```

Comment:
is MongoDB smart enough to use the index here? Even if so ... why clone the whole struct and not use the contents of the index? (the struct might get bigger in the future)